src/main/java/org/ossgang/commons/utils/GeneralDecimalFormat.java

```diff
@@ -0,0 +1,77 @@
+package org.ossgang.commons.utils;
+
+import static java.lang.Math.pow;
+
+import java.text.DecimalFormat;
+import java.text.FieldPosition;
+import java.text.NumberFormat;
+import java.text.ParsePosition;
+
+/**
+ * A Decimal Format that behaves similar to the format string "%g" ("general"): if the number is representable with
+ * a small number of digits, using fixed-point formatting, use fixed point. If it is too small or too big, fall back to
+ * scientific format.
+ */
+public class GeneralDecimalFormat extends NumberFormat {
+    private final DecimalFormat fixedFormat;
+    private final DecimalFormat exponentialFormat;
+    private final double lowExponentialThreshold;
+    private final double highExponentialThreshold;
+
+    public GeneralDecimalFormat(DecimalFormat fixedFormat, DecimalFormat exponentialFormat,
+            double lowExponentialThreshold, double highExponentialThreshold) {
+        if (lowExponentialThreshold <= 0 || highExponentialThreshold <= 0) {
+            throw new IllegalArgumentException("Thresholds must be > 0!");
+        }
+        this.fixedFormat = fixedFormat;
+        this.exponentialFormat = exponentialFormat;
+        this.lowExponentialThreshold = lowExponentialThreshold;
+        this.highExponentialThreshold = highExponentialThreshold;
+    }
+
+    public GeneralDecimalFormat(String pattern, double lowExponentialThreshold, double highExponentialThreshold) {
+        this(new DecimalFormat(pattern), new DecimalFormat(pattern + "E0"), lowExponentialThreshold,
+                highExponentialThreshold);
+    }
+
+    public GeneralDecimalFormat(int integerDigits, int fractionDigits) {
```

Comment:
consider providing factory methods with proper names. It is not obvious with two ints as parameters .. Could one consider something like?:

```
GeneralDecimalFormat.create().integerDigits(...).fractionDigits(...).
```

Reply:
I was thinking of this ... I agree a fluent API would be nicer, the only reason why I did it with the constructors is to be in line with JDK `DecimalFormat` APIs. But that is of course not a very strong argument.